feat: Allow to dismiss contact unknown callout

aleksandernsilva opened the pull request:
## Proposed changes (including videos or screenshots)

This PR adds the action to dismiss the contact unknown callout. The decision will be stored _(for the duration of the browser session)_ by contact, meaning subsequent access to rooms with the same contact won't display the callout. 

#### Before
![image](https://github.com/user-attachments/assets/a3d4e1f1-1299-4c80-8780-5c4f48749060)

#### After
<img width="702" alt="Screenshot 2025-04-04 at 11 14 26" src="https://github.com/user-attachments/assets/4fb6b16b-ccf0-4c18-acfe-935247ef3e43" />

## Issue(s)
[CTZ-32](https://rocketchat.atlassian.net/browse/CTZ-32)

## Steps to test or reproduce
- Access a workspace 
- Make sure it has single contact id addon
- Access a livechat room
- Contact unknown callout should be visible

## Further comments
<!-- If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc... -->


[CTZ-32]: https://rocketchat.atlassian.net/browse/CTZ-32?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

ggazzo commented:
> The decision will be stored by contact, meaning subsequent access to rooms with the same contact won't display the callout.
this is not exactly true right? you are not actually storing forever its only a local thing.

btw: is this never going to be removed?

dougfabris commented on apps/meteor/client/views/room/composer/ComposerOmnichannel/ComposerOmnichannelCallout.tsx:
```diff
@@ -37,14 +30,14 @@ const ComposerOmnichannelCallout = () => {
 
 	return (
 		<Callout
+			data-qa-id='contact-unknown-callout'
```
really? 

```suggestion

```

dougfabris commented on apps/meteor/client/views/room/composer/ComposerOmnichannel/ComposerOmnichannelCallout.tsx:
```diff
@@ -53,20 +46,11 @@ const ComposerOmnichannelCallout = () => {
 				</ButtonGroup>
 			}
 		>
-			{shouldShowSecurityRoute ? (
```
why we won't display anymore the option to enable the verification feature?

aleksandernsilva replied:
Even though we had the appropriate permission checks, the link was mostly relevant to admins/managers. 
The goal is to reduce the callout information to what's relevant within the conversation context, while activation/setup details will be handled in a future recommended actions page.

aleksandernsilva commented:
@ggazzo I updated the PR description to be more precise. I also spoke with product and storing the decision for the duration of the browser session meets our needs. So, I updated the logic to use session storage instead of local storage. This ensures the information is not stored indefinitely and gives users more opportunities to change their minds if they choose to.

dougfabris commented on apps/meteor/tests/e2e/page-objects/fragments/home-content.ts:
thats what I'm talking about man 👍 
help me to spread the semantic word

dougfabris reviewed (approved):
LGTM!
